controller/config_test.go

```diff
@@ -396,6 +396,33 @@ var newConfigTests = []struct {
 		controller.ControllerName: "is_invalid",
 	},
 	expectError: `controller-name value must be a valid controller name.*`,
+}, {
```

Comment:
maybe add a test case, where `controller.SSHServerPort`  not in the `controller.Config` map.. 

also, missing tests for the concurrency config